### PR TITLE
Adds error handling improvements

### PR DESCRIPTION
**Pull request by gruyaume**
Error handling improvements:
- Improves error handling to print **only** the error message instead of the whole Python Traceback errors.
- Creates new error class for operator provided arguments.
- Improves and standardised error messages.

### CODE REVIEW

File: magma_access_gateway_installer/__init__.py [hunk 1]
```diff
@@ -112,35 +122,44 @@ def cli_arguments_parser(cli_arguments):
...
     validate_arbitrary_dns(args)
     validate_custom_sgi_and_s1_interfaces(args, network_interfaces)
 
 
-def validate_custom_sgi_and_s1_interfaces(args, network_interfaces: List[str]):
+def validate_custom_sgi_and_s1_interfaces(args: argparse.Namespace, network_interfaces: List[str]):
     if args.sgi or args.s1:
         if not args.sgi or args.sgi not in network_interfaces:
-            raise ValueError("Invalid or empty SGi interface specified! Exiting...")
+            raise ArgumentError("Invalid or empty sgi argument. It must match an interface name.")
```
Comment (Gmerold) [inline magma_access_gateway_installer/__init__.py:135]:
```suggestion
            raise ArgumentError("Invalid or empty --sgi argument. It must match an interface name.")
```

Comment (Gmerold) [reply magma_access_gateway_installer/__init__.py:135]: Just to be consistent with the other messages (like _--gw-ip-address_ or _--dns_).

File: magma_access_gateway_installer/__init__.py [hunk 2]
```diff
@@ -112,35 +122,44 @@ def cli_arguments_parser(cli_arguments):
...
         if not args.s1 or args.s1 not in network_interfaces:
-            raise ValueError("Invalid or empty S1 interface specified! Exiting...")
+            raise ArgumentError("Invalid or empty s1 argument. It must match an interface name.")
```
Comment (Gmerold) [inline magma_access_gateway_installer/__init__.py:137]:
```suggestion
            raise ArgumentError("Invalid or empty --s1 argument. It must match an interface name.")
```

Comment (Gmerold) [reply magma_access_gateway_installer/__init__.py:137]: As above.

File: magma_access_gateway_installer/__init__.py [hunk 3]
```diff
@@ -112,35 +122,44 @@ def cli_arguments_parser(cli_arguments):
...
-        raise ValueError("Upstream router IP for SGi interface is missing! Exiting...")
+        raise ArgumentError("--gw-ip-address is missing.")
     elif not args.ip_address and args.gw_ip_address:
-        raise ValueError("SGi interface IP address is missing! Exiting...")
+        raise ArgumentError("--ip-address is missing.")
     elif args.ip_address and args.gw_ip_address:
         try:
             ip_network(args.ip_address, False)
+        except ValueError:
+            raise ArgumentError(f"Invalid IP address provided for ip-address ({args.ip_address}).")
```
Comment (Gmerold) [inline magma_access_gateway_installer/__init__.py:159]:
```suggestion
            raise ArgumentError(f"Invalid IP address provided for --ip-address ({args.ip_address}).")
```

File: magma_access_gateway_installer/__init__.py [hunk 4]
```diff
@@ -112,35 +122,44 @@ def cli_arguments_parser(cli_arguments):
...
+        try:
             ip_address(args.gw_ip_address)
-        except Exception as e:
-            raise e
+        except ValueError:
+            raise ArgumentError(
+                f"Invalid IP address provided for gw-ip-address ({args.gw_ip_address})."
```
Comment (Gmerold) [inline magma_access_gateway_installer/__init__.py:164]:
```suggestion
                f"Invalid IP address provided for --gw-ip-address ({args.gw_ip_address})."
```

File: magma_access_gateway_installer/__init__.py [hunk 5]
```diff
@@ -112,35 +122,44 @@ def cli_arguments_parser(cli_arguments):
...
+def validate_arbitrary_dns(args: argparse.Namespace):
     if not args.dns:
-        raise ValueError("--dns flag specified, but no addresses given! Exiting...")
+        raise ArgumentError("--dns flag specified, but no addresses given.")
     for dns_ip in args.dns:
-        ip_address(dns_ip)
+        try:
+            ip_address(dns_ip)
+        except ValueError:
+            raise ArgumentError(f"Invalid IP address provided in dns list ({dns_ip}).")
```
Comment (Gmerold) [inline magma_access_gateway_installer/__init__.py:147]:
```suggestion
            raise ArgumentError(f"Invalid IP address provided for --dns ({dns_ip}).")
```

File: magma_access_gateway_installer/__init__.py [hunk 6]
```diff
@@ -31,44 +32,53 @@ def main():
...
-    if network_configurator.reboot_needed:
-        installation_service_creator.create_magma_agw_installation_service()
-        installation_service_creator.create_magma_agw_installation_service_link()
-        logger.info(
-            "Rebooting system to apply pre-installation changes...\n"
-            "Magma AGW installation process will be resumed automatically once machine is back up."
+    try:
+        validate_args(args, network_interfaces)
+    except ArgumentError as e:
+        print(e.message)
```
Comment (sanchezfdezjavier) [inline magma_access_gateway_installer/__init__.py:38]: Why are we using `print` instead of logger?

Comment (Gmerold) [reply magma_access_gateway_installer/__init__.py:38]: Because logger is configured to log to the journal.

Comment (gruyaume) [reply magma_access_gateway_installer/__init__.py:38]: It's a good point though, we might want to use logging for both writing to the journal and printing, I'll check into this

Comment (gruyaume) [reply magma_access_gateway_installer/__init__.py]: Now fixed

**Review by Gmerold — APPROVED** [review]
One hell of an improvement ;)

**Review by sanchezfdezjavier — APPROVED** [review]

**Pull request merged**